### DISCUSSION TITLE
TreeItem::draw_item_content panics

### DISCUSSION THREAD

**Discussion by abau** [fltk-rs/fltk-rs#1406] [Q&A] [answered]
Hi,

the following panics with `'assertion failed: self.is_derived'` when calling `draw_item_content`:

```
use fltk::{app, prelude::*, window::Window, tree::*};

fn main() {
    let app = app::App::default();
    let mut wind = Window::new(100, 100, 400, 300, "Hello from rust");
    let mut tree = Tree::new(0,0,400,300, None);
    let mut item = tree.add("foo/bar").unwrap();

    item.draw_item_content(|_item, _render| 0);

    wind.end();
    wind.show();
    app.run().unwrap();
}
```

Is there something wrong with this code?

**Comment by MoAlyousef** [comment, marked as answer]
TreeItem’s that can draw are specially built to do so. The default item’s generated by `add` don’t support it. So this is intended behavior. 

You would use TreeItem::new(), set the draw_item_content callback then Tree::add_item(). 
A sample code can be found here:
https://docs.rs/fltk/latest/fltk/tree/struct.TreeItem.html#method.draw_item_content
